### ISSUE TITLE
Cannot speak in channel

### ISSUE THREAD

**Issue by Drarok** [wee-slack/wee-slack#656]
I've just installed this plugin into WeeChat 1.5, but I'm unable to say anything in channels I'm in. I can see messages, and the user list… I just can't talk!

The only clue I have is this in buffer 1:
`weechat     | python: stdout/stderr: WS ERROR`

**Comment by trygveaa** [comment]
Does this happen every time you load wee-slack?

Can you try the `debug-websocket` branch? You can download that version of the script from https://raw.githubusercontent.com/wee-slack/wee-slack/debug-websocket/wee_slack.py if you don't have a git clone.

**Comment by angezanetti** [comment]
I have the same error, I tried to use your script but I don't have more info:
```
08:52:15           weechat    ╡ python: stdout/stderr (slack): WS ERROR
08:52:57           weechat    ╡ Option changed: plugins.var.python.slack.debug_mode = "on"
08:53:03           weechat    ╡ Option changed: plugins.var.python.slack.debug_level = "2"
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
08:53:17           weechat    ╡ python: stdout/stderr (slack): WS ERROR
```
I'm running debian 7.11 with weechat 2.3

**Comment by trygveaa** [comment]
@angezanetti: Are you sure you are running that branch? You have to reload the script after changing to it.

**Comment by angezanetti** [comment]
I didn't use the branch, i wget the file you linked above and use it a weechat.py, then `/python reload`. 

```
11:06:12           weechat    ╡ python: unloading script "colorize_nicks"
11:06:12           weechat    ╡ python: unloading script "go"
11:06:12           weechat    ╡ python: unloading script "slack"
11:06:12           weechat    ╡ python: loading script "/home/xavier/.weechat/python/autoload/wee_slack.py"
11:06:12           weechat    ╡ python: registered script "slack", version 2.2.0 (Extends weechat for typing notification/search/etc on slack.com)
11:06:12           weechat    ╡ python: loading script "/home/xavier/.weechat/python/autoload/colorize_nicks.py"
11:06:12           weechat    ╡ python: registered script "colorize_nicks", version 26 (Use the weechat nick colors in the chat area)
11:06:12           weechat    ╡ python: loading script "/home/xavier/.weechat/python/autoload/go.py"
11:06:12           weechat    ╡ python: registered script "go", version 2.5 (Quick jump to buffers)
11:06:13 blupods.slack.com    ╡ Connected to Slack
11:06:13 blupods.slack.com    ╡ Websocket URL        wss://cerberus-xxxx.lb.slack-msgs.com/websocket/....
11:06:13 blupods.slack.com    ╡ User name            xavier
11:06:13 blupods.slack.com    ╡ User ID              UBEEJXXXX
11:06:13 blupods.slack.com    ╡ Team name            xxxx
11:06:13 blupods.slack.com    ╡ Team domain          xxxx
11:06:13 blupods.slack.com    ╡ Team id              TXXXXXXXX
11:06:27           weechat    ╡ python: stdout/stderr (slack): WS ERROR
```

**Comment by trygveaa** [comment]
Are you sure you put the script in the right place? Note that it says that it loaded it from the autoload directory, which may be a different script from the one in the python directory, unless it's symlinked. You can also try loading the script by path with `/python load <path>` (you will have to do `/python unload slack` first).

The reason I'm saying this is that I've added a colon and some more info after "WS ERROR" in that version, so it will never print just "WS ERROR".

**Comment by angezanetti** [comment]
Ok, sorry about that i'm a noob on Weechat :)
Here is the debug:
`13:11:55           weechat    ╡ WS ERROR: team: xxxx.slack.com, exception type: AttributeError, data: {"text": "hey", "type": "message", "user": "UBEEJXXXX", "channel": "DBCNxxxx", "id": 1}, exception arguments: ("'SlackTeam' object has no attribute 'ws'",)`

**Comment by trygveaa** [comment]
Great! I'll take a look into it.

**Comment by angezanetti** [comment]
The stack trace: 
```
 WS ERROR STACK: ['  File "/home/xavier/wee_slack.py", line 126, in wrapper\n    return f(*decode_from_utf8(args), **decode_from_utf8(kwargs))\n', '  File "/home/xavier/wee_slack.py", line 752, in buffer_input_callback\n    channel.send_message(data)\n', '  File "/home/xavier/wee_slack.py", line 1636, in send_message\n
                              ╡ self.team.send_to_websocket(request)\n', '  File "/home/xavier/wee_slack.py", line 1247, in send_to_websocket\n    w.prnt("", "WS ERROR STACK: {}".format(traceback.format_stack()))\n']
```

**Comment by trygveaa** [comment]
@angezanetti: I've updated the `debug-websocket` branch. Can you try again and see if you get a new error message?

**Comment by trygveaa** [comment]
@Killamus: Really? The only change that branch does apart from printing some errors, is to prevent sending messages over 16000 characters (including metadata). Do you get messages in the core buffer saying "Message to ... too long (... chars), not sending"? Do you get any other messages printed there?

**Comment by Killamus** [comment]
@trygveaa Yep, I noticed that too - I went and read through the diff to see if I could see the change. I don't get any error messages outside of bad commands. I'm not sure why it's working now.

**Comment by trygveaa** [comment]
@Killamus: And you get the error with the most recent master as of now? Still getting it if you change back to that?

**Comment by Forside** [comment]
I have the same error when using weechat in Cygwin. WS ERROR as soon as I write a message. Here is the stack trace from current debug-websocket branch (sensitive data replaced with XXXX):

```
weechat | WS ERROR: team: XXXX, exception type: AttributeError, data: {"text": "test",
        | "type": "message", "user": "XXXX", "channel": "XXXX", "id": 1}, exception arguments:
        | ("'SlackTeam' object has no attribute 'ws'",)
weechat | python: stdout/stderr: (slack):   File "/home/JonasH/.weechat/python/autoload/wee_slack.py",
        | line 126, in wrapper
weechat | python: stdout/stderr: (slack):     return f(*decode_from_utf8(args),
        | **decode_from_utf8(kwargs))
weechat | python: stdout/stderr: (slack):   File "/home/JonasH/.weechat/python/autoload/wee_slack.py",
        | line 743, in buffer_input_callback
weechat | python: stdout/stderr: (slack):     channel.send_message(data)
weechat | python: stdout/stderr: (slack):   File "/home/JonasH/.weechat/python/autoload/wee_slack.py",
        | line 1621, in send_message
weechat | python: stdout/stderr: (slack):     self.team.send_to_websocket(request)
weechat | python: stdout/stderr: (slack):   File "/home/JonasH/.weechat/python/autoload/wee_slack.py",
        | line 1235, in send_to_websocket
weechat | python: stdout/stderr: (slack):     traceback.print_stack()
```

Also after turning on debug mode, I get this message every second in the main weechat buffer:
`Exception in connect: error: (22, 'Invalid argument')`
It stops as soon as I tried sending a message. As I suspected something going wrong with websocket-client on Windows, I tried running weechat in Ubuntu on Windows and it works perfectly fine!

**Comment by trygveaa** [comment]
I'm guessing the error most of you here have is the same as #676, so I'm closing this issue as a duplicate (yes, this issue is older, but I'm keeping the other because it has a much clearer description and discussion).

If you believe you have a different error in the one in that issue, please enable debug mode (`/set plugins.var.python.slack.debug_mode true`), copy the error output you get in the core buffer and the debug buffer, and comment here.

The error @Forside had seems to be a different one. Not sure what that is, but seems he got it resolved.

**Comment by trygveaa** [comment]
@Forside: Regarding your issue in cygwin with `error: (22, 'Invalid argument')`. Another person asked about the same issue on irc and after some debugging we figured out that it has to do with how the socket is created in websocket-client. It seems to me like a bug in cygwin, but a workaround can be used. Turns out there already is an open PR with the workaround which fixes the issue: https://github.com/websocket-client/websocket-client/pull/547
